overlord/auth/auth_test.go

```diff
@@ -312,15 +322,31 @@ func (as *authSuite) TestGetAuthenticatorFromUser(c *C) {
 }
 
 func (as *authSuite) TestAuthenticatorSetHeaders(c *C) {
+	root, err := as.makeTestMacaroon()
+	c.Check(err, IsNil)
+	discharge, err := as.makeTestDischarge()
+	c.Check(err, IsNil)
+
+	serializedMacaroon, err := auth.MacaroonSerialize(root)
+	c.Check(err, IsNil)
+	serializedDischarge, err := auth.MacaroonSerialize(discharge)
+	c.Check(err, IsNil)
+
 	as.state.Lock()
-	user, err := auth.NewUser(as.state, "username", "macaroon", []string{"discharge"})
+	user, err := auth.NewUser(as.state, "username", serializedMacaroon, []string{serializedDischarge})
 	as.state.Unlock()
 	c.Check(err, IsNil)
 
 	req, _ := http.NewRequest("GET", "http://example.com", nil)
 	authenticator := user.Authenticator()
 	authenticator.Authenticate(req)
 
+	// discharge macaroon should be bound to the root macaroon
+	discharge.Bind(root.Signature())
+	serializedPreparedDischarge, err := auth.MacaroonSerialize(discharge)
+	c.Check(err, IsNil)
+
 	authorization := req.Header.Get("Authorization")
-	c.Check(authorization, Equals, `Macaroon root="macaroon", discharge="discharge"`)
+	expected := fmt.Sprintf(`Macaroon root="%s", discharge="%s"`, serializedMacaroon, serializedPreparedDischarge)
```

Comment:
This sort of test is a bit fragile. We're basically doing the exact same logic on both ends. In simple terms, this is equivalent to doing `foo(x) == foo(x)`. It'll tend to be true, but doesn't give us some meaningful certainty that foo is not completely broken. Imagine for example that out of bad luck we happened to update our macaroon package to something that was returning complete garbage out of Serialize. We'd not notice, because we're challenging it against itself.

Can we do something to improve the situation in those terms?